### ISSUE TITLE
[BUG]: Depth setting for Bulk Link Scraper has no effect

### ISSUE THREAD

**Issue by mPyKen** [Mintplex-Labs/anything-llm#1997]
### How are you running AnythingLLM?

Kubernetes

### What happened?

Bulk Link Scraper discovers more links than there are on the page even though depth is set to 1.

### Are there known steps to reproduce?

Open `https://www.google.com` on your browser and check that the html has 32 occurances of `href`.
Start Bulk Link Scraper on google with `depth = 1` and ~`Max Links = #occurances of href + 1 = 33`~ `Max Links = 100`.
Then check logs:
```
[collector] info: Discovering links...
[collector] info: Found 100 links to scrape.
[collector] info: Starting bulk scraping...
```

I can try to fix this if contributions are welcome

**Comment by timothycarambat** [comment]
its +1 because the root site counts as a link so that we add the origin site as well.

**Comment by mPyKen** [comment]
Fair enough, +1 makes sense. Let me update my reproduction steps: Is there any reason it detects more links than 33 when I set `Max Links = 100` and `depth = 1`?
```
[collector] info: Discovering links...
[collector] info: Found 100 links to scrape.
[collector] info: Starting bulk scraping...
[collector] info: Scraping 1/100: https://www.google.com/imghp
...
```

**Comment by timothycarambat** [comment]
@shatfield4 any ideas?

**Comment by mPyKen** [comment]
Hmm to me, it looks like depth is indeed being ignored in [discoverLinks](https://github.com/Mintplex-Labs/anything-llm/blob/3856a4d19cf925ff8177e3fcc122eeedb0198a5f/collector/utils/extensions/WebsiteDepth/index.js#L21-L40) (condition `currentLevel < depth` will always be met). If it's ok I can try to fix this and create a PR

**Comment by mPyKen** [comment]
Please let me know if this is a won't fix, help wanted or confirmation needed

**Comment by shatfield4** [comment]
@mPyKen I just created a fix for this in #2096, thank you for pointing this out!
